src/server/pps/server/api_server.go

```diff
@@ -1770,6 +1774,11 @@ func (a *apiServer) validatePipelineRequest(request *pps.CreatePipelineRequest)
 		return errors.Errorf("pipeline name is %d characters long, but must have at most 63: %q",
 			len(request.Pipeline.Name), request.Pipeline.Name)
 	}
+	// TODO(CORE-1489): Remove dependency of name length on Kubernetes
+	// resource naming convention.
+	if max := kubernetesNameLimit - len("-v99"); len(request.Pipeline.Name)+len(request.Pipeline.Project.GetName())+1 > max {
```

Comment:
We should figure out some way here to invoke ppsutil.PipelineRCName here, and then compare the length of the result to kubernetesNameLimit directly.  The name of the service/RC should also end up in the error message, as well.

I'd just `pps.PipelineInfo{Pipeline: request.Pipeline, Version: 99}` to get something you call PipelineRCName with.

As written, this is too action-at-a-distance-y.  The +1 must be the - separating project and pipeline, but nobody is going to realize that when they go to change the names.